ArcGISRuntimeSDKQt_QMLSamples/Routing/RouteAroundBarriers/RouteAroundBarriers.qml

```diff
@@ -0,0 +1,356 @@
+    property bool addStops: false
+    property bool addBarriers: false
+    property bool findBestSeq: false
+    property bool preserveFirstStp: false
+    property bool preserveLastStp: false
+
+    MapView {
+        id: mapView
+        anchors.fill: parent
+
+        Map {
+            BasemapStreets {}
+
+            ViewpointCenter {
+                Point {
+                    x: -117.1750
+                    y: 32.727
+                    SpatialReference {
+                        wkid: 4326
+                    }
+                }
+                targetScale: 40000
+            }
+        }
+
+        GraphicsOverlay {
+            id: stopsOverlay
+        }
+
+        PictureMarkerSymbol {
+            id: pinSymbol
+            url: dataPath + "/orange_symbol.png"
+        }
+
+        GraphicsOverlay {
+            id: routeOverlay
+            renderer: SimpleRenderer {
+                SimpleLineSymbol {
+                    style: Enums.SimpleLineSymbolStyleSolid
+                    color: "blue"
+                    width: 3
+                }
+            }
+        }
+
+        GraphicsOverlay {
+            id: barriersOverlay
+        }
+
+        SimpleFillSymbol {
+            id: barrierSymbol
+            style: Enums.SimpleFillSymbolStyleDiagonalCross
+            color: "red"
+            outline: SimpleLineSymbol {
+                style: Enums.SimpleLineSymbolStyleNull
+            }
+        }
+
+        RouteTask {
+            id: routeTask
+            url: routeTaskUrl
+            Component.onCompleted: {
+                load();
+            }
+
+            onLoadStatusChanged: {
+                if (loadStatus === Enums.LoadStatusLoaded) {
+                    createDefaultParameters();
+                }
+            }
+
+            onCreateDefaultParametersStatusChanged: {
+                if (createDefaultParametersStatus === Enums.TaskStatusCompleted) {
+                    routeParameters = createDefaultParametersResult;
+                    routeParameters.returnStops = true;
+                    routeParameters.returnDirections = true;
+                }
+            }
+
+            function createAndDisplayRoute() {
+                if (stopsList.length > 1) {
+                    routeOverlay.graphics.clear();
+
+                    routeParameters.setStops(stopsList);
+                    routeParameters.setPolygonBarriers(barriersList);
+                    routeParameters.findBestSequence = findBestSeq;
+                    routeParameters.preserveFirstStop = preserveFirstStp;
+                    routeParameters.preserveLastStop = preserveLastStp;
+                    solveRoute(routeParameters);
+                }
+            }
+
+            onSolveRouteStatusChanged: {
+                if (solveRouteStatus === Enums.TaskStatusCompleted) {
+                    if (solveRouteResult.routes.length === 0) {
+                        console.log("No routes found");
```

Comment:
this won't show up in the UI for users who aren't running qt creator. maybe pop up a message box or something to give the feedback?